src/hooks/useAppState/middleware.js

```diff
@@ -60,6 +60,13 @@ const applyMiddleware = (dispatch, getState) => (action) => {
       break;
     }
 
+    case types.NEW_LIST: {
+      const { key, name } = data;
+      createList(name)
+        .then((id) => dispatch({ type: types.STORE_LIST_ID, data: { key, id } }));
+      break;
+    }
```

Comment:
When a new list is created we call the API to create the list with that name, this gives us an ID which we store so we can changes later to that list (e.g. changing its name)